### PR TITLE
[annotationdb] update annotation db docs page for deploy

### PR DESCRIPTION
**Pull request by pwc2**
Changes to make sure that only the annotation datasets are visible on the docs page, now that the `datasets.json` config file contains all available datasets.

Overview:

- In `datasets.json`, moved "key_properties" inside an "annotation_db" field, like `"annotation_db":  {"key_properties":  []}`, so that only the datasets with the "annotation_db" key are shown in the annotation DB docs page. Removed "key_properties" from non-annotation datasets.

- Minor reformatting changes to docs page, added a reference genome column to the HTML table.

- Updated deploy script to reflect the filename change from `annotation_db.json` to `datasets.json`.

- Modified checks for keys in dicts from `assert key in doc, doc` to `assert key in doc` in `DatasetVersion.from_json()` and `Dataset.from_name_and_json()`. Since the `doc` that is passed to these methods from the checked in JSON file is just a dict like `doc = {"annotation_db": {"key_properties": [...]}, "description": ..., "url":  ..., "versions": [...]}` this seems to work fine. Let me know if `key in doc, doc` form was used for other reasons I've overlooked.

### CODE REVIEW

File: hail/python/hail/experimental/db.py
Comment (danking) [inline hail/python/hail/experimental/db.py:37]: In Python, the `, ...` provides more detail. For example:
```
In [1]: key = 'B'                                                                                                                                                                                                                                                               

In [2]: doc = {'A': 1}                                                                                                                                                                                                                                                          

In [3]: assert key in doc                                                                                                                                                                                                                                                       
Traceback (most recent call last):
  File "<ipython-input-3-c6a5ebd6a2ac>", line 1, in <module>
    assert key in doc
AssertionError


In [4]: assert key in doc, doc                                                                                                                                                                                                                                                  
Traceback (most recent call last):
  File "<ipython-input-4-28f2c139973f>", line 1, in <module>
    assert key in doc, doc
AssertionError: {'A': 1}

```

Let's keep them

**Review by danking — CHANGES_REQUESTED** [review]
Looks great!

File: hail/python/hail/experimental/db.py
Comment (pwc2) [reply hail/python/hail/experimental/db.py]: ah gotcha, just put them back in

**Review by danking — APPROVED** [review]
Looks good!

In the future, just dismiss my review, a PR assigned to me without a review from me will show up in my scorecard: https://scorecard.hail.is.

**Pull request merged**
